### DISCUSSION TITLE
how to launch yarpdataplayer

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#638] [Q&A] [answered]
I have a yarpdatadumper dataset from icub arms, and i want to play it in yarpdataplayer (under icub eth connected), but after i launched yaro server and yarpmanager, run yarprobotinterface,  and then select the directory from File->Open Directory, as below,
![2023-06-16_14-41](https://github.com/robotology/community/assets/7019692/4ea826c8-23d8-44a9-aff0-3ed39507d412)

and i clicked the run button, nothing happened. Is there any problem about my operation?
And another question is that can i launch yarpdataplayer in function call rather than on gui.

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

Can you share here an archive of the data you recorded?

Also, did you try this [hands-on example](https://yarp.it/latest/group__yarpdataplayer__example.html) on the `yarpdataplayer`?

>And another question is that can i launch yarpdataplayer in function call rather than on GUI.

Yep, although I'm not an expert.

In the documentation, it's written to launch the module with `--hidden` command line param and then to operate via the relative RPC port.

The commands are IDL-generated and documented at:
- https://github.com/robotology/yarp/blob/master/src/yarpdataplayer-console/idl_generated_code/yarpdataplayer_console_IDL.h

**Reply by jimingda** [reply]
Thanks @pattacini 
I tried the[ hands on example](https://yarp.it/latest/group__yarpdataplayer__example.html),  launched the yarpdataplayer example, chose and opened  the testData_20120803_095402 on the yarpdataplayer, the dataset can be played up to100%, but robot on /icubgui didn't move and no pic showed on two cam viewers.

**Reply by pattacini** [reply]
Did you connect the ports?

**Reply by jimingda** [reply]
When i clicked the connectall button on yarpmanager gui, err msg showed as below, shall i modify the configuration somewhere？
![2023-06-21_00-26](https://github.com/robotology/community/assets/7019692/84c398ff-0f82-4cb1-95be-b58f47c897bd)

**Reply by Nicogene** [reply]
Hi @jimingda, as you can see it is trying to connect the `/icub/camCalib*` ports but they are not opened,  you have to check the names of the ports opened by `yarpdataplayer` and edit the "from" field w/  the correct names.

**Reply by jimingda** [reply]
It seems the from ports don't need change and i can connect most of them, but still some disconnected like inertial and right_arm. And nothing happened on iCubGui after i ran the test dataset on yarpdataplayer. Shall i change the carrier on yarpmanager from udp to tcp?
![2023-06-22_07-31](https://github.com/robotology/community/assets/7019692/b74d4446-9616-4629-ac42-f099fba5cfac)

**Reply by pattacini** [reply]
The name of the inertial port has changed into `/iCubGui/inertials/measures:i`. I will patch the XML app, but you can already fix the name yourself, @jimingda.

The other ports remain red probably because there're address conflicts. Try to run the iCubGui from the console to check the output.

**Reply by pattacini** [reply]
Related PR:
- https://github.com/robotology/yarp/pull/2979

**Reply by jimingda** [reply]
What is the file path of these default demos on yarpmanager, i think the xml file is within these path.
If i modify the xml, shall i rebuild the yarpdataplayer_example to enable it.
And how to run the iCubGui from the console?  can i run the yarpdataplayer example on icub rather than iCubGui.

**Reply by pattacini** [reply]
Hi @jimingda 

>What is the file path of these default demos on yarpmanager, i think the xml file is within these path.

If you unfold the application on the left panel, you'll find out its path as in the screenshot below.

<img width="900" alt="Screenshot 2023-06-22 141800" src="https://github.com/robotology/community/assets/3738070/f5e7c57f-b60f-43f5-8c46-4ad8c5163450">

Also, depending on how you configured the applications to open/edit XML files, it suffices to double-click on the path to open it in the default editor.

>If i modify the xml, shall i rebuild the yarpdataplayer_example to enable it.

Once edited, you can simply relaunch `yarpmanager` and the app is ready.
Regarding the inertials, we realized that they need to be removed from the XML outright, as explained in https://github.com/robotology/yarp/pull/2979#issuecomment-1602308838.

>And how to run the iCubGui from the console?

A console is a terminal, so launch it from there by typing `iCubGui`.

> can i run the yarpdataplayer example on icub rather than iCubGui.

Nope, because the `iCubGui` is not a robot or a simulated robot. As [I already suggested you do](https://github.com/orgs/robotology/discussions/635#discussioncomment-6161516), you ought to be using [`wholeBodyPlayer`](https://robotology.github.io/robotology-documentation/doc/html/group__wholeBodyPlayer.html).

**Reply by jimingda** [reply]
Without modify the yarpdataplayerExample.xml (i.e  remove the below part), i can still play the testdata, as the pic attached.
But when i replace the testdata with my dump leftArm data(tleftarm moving log), the data can be import correctly, and the robot on icubGui didn't move.

```
 52     <connection>
 53         <from>/icub/inertial</from>
 54         <to>/iCubGui/inertial:i</to>
 55         <protocol>udp</protocol>
 56     </connection>
```

![2023-06-27_17-42](https://github.com/robotology/community/assets/7019692/661373ee-d19d-4d1c-ab4d-958813c10391)

![2023-06-27_18-06](https://github.com/robotology/community/assets/7019692/63077731-8d8c-40d3-bda3-af6089288bee)

Shall i modify the Speed on yarpdataplayer or the FPS on /iCubGui

**Reply by pattacini** [reply]
>Shall i modify the Speed on yarpdataplayer or the FPS on /iCubGui

I don't expect that this would sort out the problem.

You may consider attaching here a zip containing your recorded sequence so that we can inspect it.

**Reply by jimingda** [reply]
Thanks @pattacini   The problem is fixed.

**Reply by pattacini** [reply]
That's great! You may mark one of the messages above as the answer then. Thanks!
